test/performance/latency-controller/README.md

```diff
@@ -1,242 +1,177 @@
-      httpGet:
-        path: /
-        port: 80
-      initialDelaySeconds: 5
-      periodSeconds: 5
-EOF
-```
+- **Multi-Cluster Support**: Monitors workloads across multiple WEC clusters simultaneously  
+- **Dynamic Resource Discovery**: Automatically discovers and monitors all configured Kubernetes resource types  
+- **Real-time Metrics**: Provides live latency measurements via Prometheus metrics  
+- **Workload Count Tracking**: Monitors the number of deployed workload objects per cluster  
+- **Status Field Detection**: Intelligently handles resources with and without status fields  
 
-## 🛠️ Build & Run
+## Prerequisites
 
-Run
+Before deploying the Latency Controller, ensure you have:
 
-```bash
-go build -o bin/latency-controller ./test/performance/latency-controller
+1. **KubeStellar Environment**: You must have an environment with KubeStellar installed; see [KubeStellar getting started](https://docs.kubestellar.io/release-0.23.1/direct/get-started/). Alternatively, you can also use KubeStellar e2e script [run-test.sh](https://github.com/kubestellar/kubestellar/blob/main/test/e2e/run-test.sh) to setup an environment.
 
-./bin/latency-controller
-```
+2. **KubeStellar Monitoring Setup**: The base monitoring infrastructure must be installed first:
+   - Setup the KS monitoring tool [Here](https://github.com/kubestellar/kubestellar/tree/main/monitoring#readme)
+   - This installs Prometheus and Grafana components
 
-## 📊 Monitoring Setup
+## Installation
 
-### Another Terminal (Prometheus)
+### Step 1: Build and Push Controller Image
 
+Build and push the controller image to your container registry:
 ```bash
-# Download and extract
-wget https://github.com/prometheus/prometheus/releases/download/v2.51.0/prometheus-2.51.0.linux-amd64.tar.gz
-tar -xzf prometheus-2.51.0.linux-amd64.tar.gz
-cd prometheus-2.51.0.linux-amd64
-```
+Build the controller image
+make docker-copy
+make docker-build IMAGE=<your-registry>/latency-controller:tag
 
-Create `prometheus.yml`:
+Push to registry
+make docker-push IMAGE=<your-registry>/latency-controller:tag
+```
 
+### Step 2: Configure Latency Controller
 ```bash
-global:
-  scrape_interval: 15s
-  evaluation_interval: 15s
-
-scrape_configs:
-  - job_name: 'kubestellar-latency'
-    static_configs:
-      - targets: ['localhost:2222']
-    metrics_path: '/metrics'
+# 1. Create latency-collector service
+sed s/%WDS%/wds1/g configuration/latency-collector-svc.yaml | kubectl -n wds1-system apply -f -
+
+# 2. Create latency-collector service monitor
+sed s/%WDS%/wds1/g configuration/latency-collector-sm.yaml | kubectl -n ks-monitoring apply -f -
 ```
 
-Run Prometheus:
+### Step 3: Deploy the Latency Controller
 
+Use the deployment script to install the controller:
 ```bash
-./prometheus \
-  --config.file=prometheus.yml \
-  --storage.tsdb.path=./data \
-  --web.listen-address=0.0.0.0:9090
+./deploy-latency-controller.sh
+--latency_controller_image "<CONTROLLER_IMAGE>"
+--binding-name "<BINDING_NAME>"
```

Comment:
Here should be `bindingPolicy` name instead of `bindingName` - bindings is referring to another object in KubeStellar.